api/v1alpha1/nonadminbackupstoragelocation_types.go

```diff
@@ -0,0 +1,104 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	velerov1 "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NonAdminBackupStorageLocationPhase is a simple one high-level summary of the lifecycle of an NonAdminBackupStorageLocation.
+// +kubebuilder:validation:Enum=New;Available;Unavailable;Created;Deleting
+type NonAdminBackupStorageLocationPhase string
+
+// NonAdminBackupStorageLocationPhase constants similar to velerov1.BackupStorageLocationPhase
+const (
+	NaBSLPhaseNew         NonAdminBackupStorageLocationPhase = "New"
+	NaBSLPhaseAvailable   NonAdminBackupStorageLocationPhase = "Available"
+	NaBSLPhaseUnavailable NonAdminBackupStorageLocationPhase = "Unavailable"
+	NaBSLPhaseCreated     NonAdminBackupStorageLocationPhase = "Created"
+	NaBSLPhaseDeleting    NonAdminBackupStorageLocationPhase = "Deleting"
+)
+
+// NonAdminBSLCondition contains addition conditions to the
+// generic ones defined as NonAdminCondition
+// +kubebuilder:validation:Enum=SecretSynced;BSLSynced
+type NonAdminBSLCondition string
+
+// Predefined NonAdminBSLConditions
+const (
+	NonAdminBSLConditionSecretSynced NonAdminBSLCondition = "SecretSynced"
+	NonAdminBSLConditionBSLSynced    NonAdminBSLCondition = "BackupStorageLocationSynced"
+)
+
+// NonAdminBackupStorageLocationSpec defines the desired state of NonAdminBackupStorageLocation
+type NonAdminBackupStorageLocationSpec struct {
+	BackupStorageLocationSpec *velerov1.BackupStorageLocationSpec `json:"backupStorageLocationSpec"`
+}
+
+// VeleroBackupStorageLocation contains information of the related Velero backup object.
+type VeleroBackupStorageLocation struct {
+	// status captures the current status of the Velero backup storage location.
+	// +optional
+	Status *velerov1.BackupStorageLocationStatus `json:"status,omitempty"`
+
+	// nacuuid references the Velero BackupStorageLocation object by it's label containing same NACUUID.
+	// +optional
+	NACUUID string `json:"nacuuid,omitempty"`
+
+	// references the Velero BackupStorageLocation object by it's name.
+	// +optional
+	Name string `json:"name,omitempty"`
+
+	// namespace references the Namespace in which Velero backup storage location exists.
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+}
+
+// NonAdminBackupStorageLocationStatus defines the observed state of NonAdminBackupStorageLocation
+type NonAdminBackupStorageLocationStatus struct {
+	// +optional
+	VeleroBackupStorageLocation *VeleroBackupStorageLocation `json:"veleroBackupStorageLocation,omitempty"`
+
+	Phase      NonAdminBackupStorageLocationPhase `json:"phase,omitempty"`
+	Conditions []metav1.Condition                 `json:"conditions,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
```

Comment:
Lets also add shortname here
`// +kubebuilder:resource:path=nonadminbackupstoragelocations,shortName=nabsl`

Reply:
Fixed, will commit soon.

Reply:
Can we close this as resolved?